stumpy/snippets.py

```diff
@@ -83,22 +94,25 @@ def _get_all_profiles(
             f"Please try `m <= len(T) // 2`."
         )
 
+    if s is not None:
+        s = min(int(s), m)
+    else:
+        percentage = np.clip(percentage, 0.0, 1.0)
+        s = min(math.ceil(percentage * m), m)
+
     right_pad = 0
+    T_subseq_isconstant = core.rolling_isconstant(T, s, T_subseq_isconstant)
     if T.shape[0] % m != 0:
         right_pad = int(m * np.ceil(T.shape[0] / m) - T.shape[0])
         pad_width = (0, right_pad)
         T = np.pad(T, pad_width, mode="constant", constant_values=np.nan)
+        T_subseq_isconstant = np.pad(
+            T_subseq_isconstant, pad_width, mode="constant", constant_values=False
+        )
 
     n_padded = T.shape[0]
     D = np.empty(((n_padded // m) - 1, n_padded - m + 1), dtype=np.float64)
 
-    if s is not None:
-        s = min(int(s), m)
-    else:
-        percentage = np.clip(percentage, 0.0, 1.0)
-        s = min(math.ceil(percentage * m), m)
-
-    T_subseq_isconstant = core.rolling_isconstant(T, s)
     M_T, Σ_T = core.compute_mean_std(T, s)
```

Comment:
Since I moved the line `T_subseq_isconstant = core.rolling_isconstant(T, s)`, I was wondering if we should move this line above as well for the sake of consistency and flow in the code.

Reply:
I think it's fine either way